pkg/vm/client.go

```diff
@@ -86,40 +93,48 @@ func (c *Client) Resume(ctx context.Context) error {
 }
 
 // Inspect return information about the vm
-func (c *Client) Inspect(ctx context.Context) (CPU, MemMib, error) {
+func (c *Client) Inspect(ctx context.Context) (VMData, error) {
 	request, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://unix/api/v1/vm.info", nil)
 	if err != nil {
-		return 0, 0, err
+		return VMData{}, err
 	}
 	request.Header.Add("content-type", "application/json")
 
 	response, err := c.client.Do(request)
 	if err != nil {
-		return 0, 0, errors.Wrap(err, "error calling machine info")
+		return VMData{}, errors.Wrap(err, "error calling machine info")
 	}
 
 	defer response.Body.Close()
 
 	if response.StatusCode != http.StatusOK {
-		return 0, 0, fmt.Errorf("got unexpected http code '%s' on machine info", response.Status)
+		body, _ := ioutil.ReadAll(response.Body)
+		response.Body.Close()
+
+		return VMData{}, fmt.Errorf("got unexpected http code '%s' on machine info, Response: %s", response.Status, string(body))
 	}
 
 	var data struct {
 		Config struct {
-			CPUs struct {
+			CPU struct {
```

Comment:
Why not use the `VMData` struct here instead of anonymous  struct then copy everything. Just make sure VMData has the right json annotiations then return data.Config directly